editor/src/core/shared/uid-utils.ts

```diff
@@ -106,17 +106,25 @@ const atoz = [
 
 // Assumes possibleStartingValue consists only of characters that are valid to begin with.
 export function generateConsistentUID(
-  existingIDs: Set<string>,
   possibleStartingValue: string,
+  ...existingIDSets: Array<Set<string>>
 ): string {
+  function alreadyExistingID(idToCheck: string): boolean {
+    for (const existingIDs of existingIDSets) {
+      if (existingIDs.has(idToCheck)) {
+        return true
+      }
+    }
+    return false
+  }
```

Comment:
You could simplify this to:
```suggestion
  function alreadyExistingID(idToCheck: string): boolean {
    return existingIDSets.some(s => s.has(idToCheck))
  }
```
I don't think this is a pattern we've used before, otherwise I'd have suggested adding a generic function to `set-utils.ts`

Reply:
That's a lot better, I wrote it longer hand at the time because I was trying to think through what I actually wanted, but that's not necessary now.